From e20af915661afa7b1aa10e7caafb83c6b9ad9a39 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Wed, 6 Mar 2019 23:49:53 +0200
Subject: [PATCH 5/5] su: Add back mistakenly removed code

Commit 71839266b7a3b18bd0824e1896c8dab7ed217508 removed this part
of the code because it was mistakenly thought as a samsung hack.

This was a misunderstanding because of the highly outdated comment,
which was referring to commit 62843e15043a07952ec2d16e50db2bbefeebb571
hacks which were later removed in 379371bcff3f18b4a16beb37a35c2b211a941781

This should fix running su when you are already root.

Change-Id: Id548cb8499eaab943c18a5ed564da8f61e4e390c
---
 su.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/su.c b/su.c
index 808be9e..8bf8b92 100644
--- a/su.c
+++ b/su.c
@@ -475,6 +475,11 @@ int su_main(int argc, char* argv[], int need_client) {
 
     ALOGE("SU from: %s", ctx.from.name);
 
+    if (ctx.from.uid == AID_ROOT) {
+        ALOGD("Allowing root.");
+        allow(&ctx, NULL);
+    }
+
     // check if superuser is disabled completely
     if (access_disabled(&ctx.from)) {
         ALOGD("access_disabled");
-- 
2.17.1

